src/common/aes/aes256_ni.c

```diff
@@ -184,18 +184,18 @@ void oqs_aes256_ctr_enc_sch_upd_blks_ni(void *schedule, uint8_t *out, size_t out
 
 	while (out_blks >= 4) {
 		__m128i nv0 = _mm_shuffle_epi8(ctx->iv, mask);
-		__m128i nv1 = _mm_shuffle_epi8(_mm_add_epi64(ctx->iv, _mm_set_epi64x(1, 0)), mask);
-		__m128i nv2 = _mm_shuffle_epi8(_mm_add_epi64(ctx->iv, _mm_set_epi64x(2, 0)), mask);
-		__m128i nv3 = _mm_shuffle_epi8(_mm_add_epi64(ctx->iv, _mm_set_epi64x(3, 0)), mask);
+		__m128i nv1 = _mm_shuffle_epi8(_mm_add_epi32(ctx->iv, _mm_set_epi64x(1, 0)), mask);
```

Comment:
Does the `_mm_set_epi64x` call also need to be switched to a `32` version to match?

Reply:
I'm not very familiar with the details of these intrinsics, so treat this as question informed by pattern match, not deep technical understanding.

Reply:
It will give the same result as _mm_set_epi32(0, i, 0, 0), where i is the amount being incremented. Since the value it is being added to has had the 1st and 2nd 64 bit registers swapped using _mm_shuffle_epi8, it is the same as adding (0, 0, 0, i) which matches the software implementation, but it may be more readable to use the _mm_set_epi32 function.